### PR TITLE
Sudoless Travis

### PR DESCRIPTION
**Pull request by dwaynebailey**
https://tools.hmcts.net/jira/browse/RDM-2288

No sudo needed, cleanup, better Gradle caching

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,22 @@
-sudo: required
 language: java
 jdk:
-- oraclejdk8
+  - oraclejdk8
+
 before_cache:
-- rm -fr $HOME/.gradle/caches/
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
```
Comment (adr1ancho) [inline .travis.yml]: May I ask is there anything else you would like to keep from $HOME/.gradle/caches

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,22 @@
-sudo: required
 language: java
 jdk:
-- oraclejdk8
+  - oraclejdk8
+
```
Comment (adr1ancho) [inline .travis.yml]: Indentation can be overridden when I was experimented with coverity

File: .travis.yml (hunk 1 above)
Comment (dwaynebailey) [reply .travis.yml]: Just going on Travis recommendations, as it is its throwing away everything, so pretty pointless.

File: .travis.yml (hunk 2 above)
Comment (dwaynebailey) [reply .travis.yml]: Not sure I'm following, this comment

**Pull request merged**